server-base/src/main/java/org/apache/kylin/rest/job/KylinHealthCheckJob.java

```diff
@@ -0,0 +1,298 @@
+
+  public static void main(String[] args) throws Exception {
+    new KylinHealthCheckJob().execute(args);
+  }
+  final KylinConfig config;
+  final BufferedLogger reporter = new BufferedLogger(logger);
+  final CubeManager cubeManager;
+
+  public KylinHealthCheckJob() {
+    this(KylinConfig.getInstanceFromEnv());
+  }
+
+  public KylinHealthCheckJob(KylinConfig config) {
+    this.config = config;
+    this.cubeManager = CubeManager.getInstance(config);
+  }
+
+  @Override
+  protected Options getOptions() {
+    Options options = new Options();
+    // TODO: Support to fix the unhealthy cube automatically
+    options.addOption(OPTION_FIX);
+    return options;
+  }
+
+  @Override
+  protected void execute(OptionsHelper optionsHelper) throws Exception {
+    logger.info("options: '" + optionsHelper.getOptionsAsString() + "'");
+    checkCubeHealth();
+  }
+
+  private void checkCubeHealth() throws Exception {
+    CubeManager cubeManager = CubeManager.getInstance(config);
+
+    List<CubeInstance> cubes = cubeManager.listAllCubes();
+    // TODO:
+
+    checkErrorMeta();
+
+    // Check if the cubeid data exist for later cube merge
+    checkSegmentHDFSPath(cubes);
+
+    // Check if the hbase table exits or online
+    checkHBaseTables(cubes);
+
+    // Check if there are holes in cube
+    // TODO: check if there are overlaps in segments of cube
+    checkCubeHoles(cubes);
+
+    // Check if there are too many segments
+    checkTooManySegments(cubes);
+
+    // Check if there are stale metadata
+    checkStaleSegments(cubes);
+
+    // Disable/Delete the out-of-date cube
+    checkOutOfDateCube(cubes);
+
+    // Check data expand rate
+    checkDataExpansionRate(cubes);
+
+    // Check auto merge param
+    checkCubeDescParams(cubes);
+
+    // ERROR history stopped build job
+    checkStoppedJob();
+
+    sendMail(reporter.getBufferedLog());
+  }
+
+  private void sendMail(String content) {
+    logger.info("Send Kylin cluster report");
+    String subject = "Kylin Cluster Health Resport of " + config.getClusterName()
+        + " on " + new SimpleDateFormat("yyyy-MM-dd", Locale.ROOT).format(new Date());
+    List<String> users = Lists.newArrayList(config.getAdminDls());
+    new MailService(config).sendMail(users, subject, content, false);
+  }
+
+  private void checkErrorMeta() {
+    reporter.log("## Checking metadata");
+
+    CubeManager cubeManager = CubeManager.getInstance(config);
+    for (String cube: cubeManager.getErrorCubes()) {
+      reporter.log("Error loading CubeDesc at " + cube);
+    }
+
+    DataModelManager modelManager = DataModelManager.getInstance(config);
+    for (String model: modelManager.getErrorModels()) {
+      reporter.log("Error loading DataModelDesc at " + model);
+    }
+  }
+
+  private void checkStoppedJob() throws Exception {
+    reporter.log("## Cleanup stopped job");
+    // TODO: Add a config
+    long outdatedJobTimeCut = System.currentTimeMillis() - 30 * 24 * 3600 * 1000L;
```

Comment:
It looks like that `30 * 24 * 3600 * 1000L` is a magic number, could you please add these in **KylinConfig**?